In [1]:
import pandas as pd
import matplotlib.pyplot as plt
#import datetime
import numpy as np

In [4]:
df2 = pd.read_csv('Airlift - case study - data set - Data set 2.csv')
df2.head()

id  warehouse_id  product_id  hours_out_of_stock  daily_sales_adj  \
0  2408059             1           9           11.765823         5.657077   
1  2454707             1          10            9.660333        11.868269   
2  2482569             1          13            0.209807         9.000000   
3  2222342             1          17           14.000000         1.997337   
4  2441527             1          24           11.766224         0.714286   

   selling_price       day    city  
0           60.0  4/1/2021  Lahore  
1           20.0  4/1/2021  Lahore  
2          111.0  4/1/2021  Lahore  
3          159.0  4/1/2021  Lahore  
4          580.0  4/1/2021  Lahore

In [5]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260504 entries, 0 to 260503
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  260504 non-null  int64  
 1   warehouse_id        260504 non-null  int64  
 2   product_id          260504 non-null  int64  
 3   hours_out_of_stock  260504 non-null  float64
 4   daily_sales_adj     204399 non-null  float64
 5   selling_price       260504 non-null  float64
 6   day                 260504 non-null  object 
 7   city                260504 non-null  object 
dtypes: float64(3), int64(3), object(2)
memory usage: 15.9+ MB


In [6]:
df2.describe()

id   warehouse_id     product_id  hours_out_of_stock  \
count  2.605040e+05  260504.000000  260504.000000       260504.000000   
mean   2.766015e+06      47.135825    4060.901633            9.740099   
std    6.490961e+05      35.413071    2982.433859            5.310028   
min    9.048000e+03       1.000000       1.000000            0.000000   
25%    2.530297e+06      35.000000    1474.000000            4.797499   
50%    2.837228e+06      38.000000    3350.000000           14.000000   
75%    3.186382e+06      68.000000    6432.000000           14.000000   
max    3.741990e+06     169.000000   11563.000000           14.992370   

       daily_sales_adj  selling_price  
count    204399.000000  260504.000000  
mean          3.946806     311.622367  
std           9.957354    1164.930764  
min           0.000000       5.000000  
25%           0.428571      75.000000  
50%           1.013274     155.000000  
75%           2.900188     325.000000  
max         223.171777   48900.000000

## Average time hours_out_of_stock is: 9.74 hours
## Max time hours_out_of_stock is: 14.99 hours

In [7]:
df2.isna().sum()

id                        0
warehouse_id              0
product_id                0
hours_out_of_stock        0
daily_sales_adj       56105
selling_price             0
day                       0
city                      0
dtype: int64

In [8]:
#convert from object to date time
df2['day'] = pd.to_datetime(df2['day'])
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260504 entries, 0 to 260503
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype         
---  ------              --------------   -----         
 0   id                  260504 non-null  int64         
 1   warehouse_id        260504 non-null  int64         
 2   product_id          260504 non-null  int64         
 3   hours_out_of_stock  260504 non-null  float64       
 4   daily_sales_adj     204399 non-null  float64       
 5   selling_price       260504 non-null  float64       
 6   day                 260504 non-null  datetime64[ns]
 7   city                260504 non-null  object        
dtypes: datetime64[ns](1), float64(3), int64(3), object(1)
memory usage: 15.9+ MB


In [10]:
df2['month'] = df2['day'].dt.month


id  warehouse_id  product_id  hours_out_of_stock  daily_sales_adj  \
0  2408059             1           9           11.765823         5.657077   
1  2454707             1          10            9.660333        11.868269   
2  2482569             1          13            0.209807         9.000000   
3  2222342             1          17           14.000000         1.997337   
4  2441527             1          24           11.766224         0.714286   

   selling_price        day    city  month  
0           60.0 2021-04-01  Lahore      4  
1           20.0 2021-04-01  Lahore      4  
2          111.0 2021-04-01  Lahore      4  
3          159.0 2021-04-01  Lahore      4  
4          580.0 2021-04-01  Lahore      4

## Data for only 1 Month (April 2021) 

In [18]:
df2.month.nunique()

1

In [19]:
df2.day.tail()

260499   2021-04-30
260500   2021-04-30
260501   2021-04-30
260502   2021-04-30
260503   2021-04-30
Name: day, dtype: datetime64[ns]

## Data for only 3 cities 

In [21]:
df2.city.unique()

array(['Lahore', 'Karachi', 'Islamabad'], dtype=object)

In [23]:
df2.city.value_counts()

Lahore       150689
Karachi       96826
Islamabad     12989
Name: city, dtype: int64

## 10 warehouses across pakistan 

In [27]:
df2.warehouse_id.nunique()

10

In [28]:
df2.warehouse_id.value_counts()

1      47596
39     34423
68     32564
35     30700
38     28347
37     26061
69     24774
67     17644
135    12989
169     5406
Name: warehouse_id, dtype: int64

## Number of products in those warehouses: 6607 

In [30]:
df2.product_id.nunique()

6607